packages/instrumentation-socket.io/src/socket.io.ts

```diff
@@ -0,0 +1,217 @@
+                }
+                if (isWrapped(moduleExports.Server.prototype.on)) {
+                    this._unwrap(moduleExports.Server.prototype, 'on');
+                }
+            },
+            [broadcastOperatorInstrumentation, namespaceInstrumentation, socketInstrumentation]
+        );
+    }
+
+    private _patchOn() {
+        const self = this;
+        return (original: Function) => {
+            return function (ev: any, originalListener: Function) {
+                if (!self._config.traceReserved && reservedEvents.includes(ev)) {
+                    return original.apply(this, arguments);
+                }
+                const wrappedListener = function () {
+                    const operation = 'on';
+                    const messageName = ev;
+
+                    const span: Span = self.tracer.startSpan(`socket.io ${operation} ${messageName}`, {
+                        kind: SpanKind.CONSUMER,
+                        attributes: {
+                            [SemanticAttributes.MESSAGING_SYSTEM]: 'socket.io',
+                            [SemanticAttributes.MESSAGING_DESTINATION]: originalListener.name,
+                            [SemanticAttributes.MESSAGING_OPERATION]: MessagingOperationValues.PROCESS,
+                        },
+                    });
+
+                    if (self._config.onHook) {
+                        safeExecuteInTheMiddle(
+                            () => self._config.onHook(span, { args: arguments }),
+                            (e) => {
+                                if (e) diag.error(`socket.io instrumentation: onHook error`, e);
+                            },
+                            true
+                        );
+                    }
+
+                    const result = context.with(setSpan(context.active(), span), () =>
+                        originalListener.apply(this, arguments)
+                    );
+                    span.end();
+                    return result;
+                };
+
+                return original.apply(this, [ev, wrappedListener]);
+            };
+        };
+    }
+
+    private _patchEmit() {
+        const self = this;
+        return (original: Function) => {
+            return function (ev: any, ...args: any[]) {
+                if (!self._config.traceReserved && reservedEvents.includes(ev)) {
+                    return original.apply(this, arguments);
+                }
+                const operation = 'emit';
+                const messageName = ev;
+                const attributes = {
+                    [SemanticAttributes.MESSAGING_SYSTEM]: 'socket.io',
+                    [SemanticAttributes.MESSAGING_DESTINATION_KIND]: 'topic',
+                    [SemanticAttributes.MESSAGING_OPERATION]: operation,
+                };
+
+                const rooms = this.rooms || this._rooms || this.sockets._rooms;
+                if (rooms && rooms.size > 0) {
```

Comment:
not sure how that would help me here.
since this `rooms` property was moved around in earlier versions I check for it all possible places instead of having a patch function per-version

Reply:
Yes that is good, I suggest to write `rooms?.size` instead.
Optional chaining was designed to replace code that check bullish values instead of `&&` operator